src/Components/Components/src/RenderTree/RenderTreeDiffBuilder.cs

```diff
@@ -35,107 +36,256 @@ public static void DisposeFrames(RenderBatchBuilder batchBuilder, ArrayRange<Ren
             int oldStartIndex, int oldEndIndexExcl,
             int newStartIndex, int newEndIndexExcl)
         {
+            // This is deliberately a very large method. Parts of it could be factored out
+            // into other private methods, but doing so comes at a consequential perf cost,
+            // because it involves so much parameter passing. You can think of the code here
+            // as being several private methods (delimited by #region) pre-inlined.
+            //
+            // A naive "extract methods"-type refactoring will worsen perf by about 10%. So,
+            // if you plan to refactor this, be sure to benchmark the old and new versions
+            // on Mono WebAssembly.
```

Comment:
Is this true even with `[MethodImpl(MethodImplOptions.AggressiveInlining)]` applied to the child methods?

Reply:
Yes, because the Mono WebAssemby interpreter isn't JITing.

Reply:
I'm all for big methods. Bring it on